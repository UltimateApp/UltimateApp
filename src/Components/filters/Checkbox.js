import React from 'react';
import { StyleSheet, View } from 'react-native';
import { CheckBox as NativeCheckbox } from 'react-native-elements';
import theme from '../../styles/theme.style';

const Checkbox = (props) => {
  return (
    <View style={styles.container}>
      <NativeCheckbox
        checked={props.active}
        onPress={props.onPress}
        key={props.title}
        title={props.title}
        containerStyle={styles.containerStyle}
        textStyle={styles.textStyle}
        checkedColor={theme.MAIN_COLOR}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexBasis: '50%',
  },
  containerStyle: {
    backgroundColor: theme.BACKGROUND_COLOR_LIGHT,
    borderWidth: 0,
    padding: 0,
  },
  textStyle: {
    textTransform: 'capitalize',
    fontWeight: 'normal',
  },
});

export default Checkbox;
