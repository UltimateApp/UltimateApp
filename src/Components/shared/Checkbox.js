import React from 'react';
import { StyleSheet, View } from 'react-native';
import { CheckBox as NativeCheckbox } from 'react-native-elements';
import theme from '../../styles/theme.style';

const Checkbox = props => {
  return (
    <View style={styles.container}>
      <NativeCheckbox
        checked={props.active}
        onPress={props.onPress}
        key={props.title}
        title={props.title}
        containerStyle={styles.containerStyle}
        wrapperStyle={styles.wrapperStyle}
        textStyle={styles.textStyle}
        checkedColor={theme.FORM_ELEMENT_COLOR}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexBasis: '50%',
  },
  containerStyle: {
    backgroundColor: theme.BACKGROUND_COLOR_LIGHT,
    borderWidth: 0,
    margin: 0,
    padding: 0,
    marginRight: 5,
  },
  wrapperStyle: {
    margin: 0,
    padding: 0,
    borderWidth: 0,
  },
  textStyle: {
    textTransform: 'capitalize',
  },
});

export default Checkbox;
