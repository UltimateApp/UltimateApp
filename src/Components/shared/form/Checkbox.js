import React, { useState } from 'react';
import { StyleSheet, TouchableOpacity, View, Text } from 'react-native';
import { CheckBox as NativeCheckbox } from 'react-native-elements';
import { useField } from 'formik';

import I18n from '../../../utils/i18n';
import theme from '../../../styles/theme.style';

const Checkbox = ({ fieldName, label, values, labels, ...props }) => {
  const [field, meta, helpers] = useField(fieldName);

  const renderCheckbox = (possibleValue, index) => {
    const selected = field.value.includes(possibleValue);

    const onPress = () => {
      const existingValueIndex = field.value.findIndex((v) => v === possibleValue);
      if (existingValueIndex === -1) {
        helpers.setValue([...field.value, possibleValue]);
      } else {
        helpers.setValue(field.value.filter((value, index) => index !== existingValueIndex));
      }
    };

    return (
      <NativeCheckbox
        checked={field.value.includes(possibleValue)}
        onPress={onPress}
        key={index}
        title={labels[index]}
        containerStyle={styles.checkbox}
        textStyle={styles.checkboxText}
        checkedColor={theme.MAIN_COLOR}
      />
    );
  };

  return (
    <View style={styles.group}>
      <Text style={styles.label}>{label}</Text>
      {meta.error && meta.touched && <Text style={styles.error}>{meta.error}</Text>}
      <View style={styles.wrapper}>{values.map(renderCheckbox)}</View>
    </View>
  );
};

export default Checkbox;

const styles = StyleSheet.create({
  group: {
    marginBottom: 10,
  },
  label: {
    marginBottom: 5,
  },
  wrapper: {
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  checkbox: {
    flexBasis: '40%',
    backgroundColor: theme.BACKGROUND_COLOR_LIGHT,
    borderWidth: 0,
    padding: 0,
  },
  checkboxText: {
    fontWeight: 'normal',
  },
  error: {
    fontStyle: 'italic',
    color: 'red',
  },
});
