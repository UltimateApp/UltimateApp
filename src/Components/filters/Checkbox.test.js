import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react-native';

import Checkbox from './Checkbox';

describe('<Checkbox />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Checkbox title="OK" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when active', () => {
    const tree = renderer.create(<Checkbox title="OK" active />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls delegates onPress prop', async () => {
    const onPress = jest.fn();

    const { getByText } = render(<Checkbox title="Easy" onPress={onPress} />);

    await fireEvent.press(getByText('Easy'));

    expect(onPress).toHaveBeenCalled();
  });
});
