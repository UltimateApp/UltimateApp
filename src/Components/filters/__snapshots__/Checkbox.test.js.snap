// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> renders correctly 1`] = `
<View
  style={
    Object {
      "flexBasis": "50%",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#ededed",
        "borderRadius": 3,
        "borderWidth": 0,
        "margin": 5,
        "marginLeft": 10,
        "marginRight": 10,
        "opacity": 1,
        "padding": 0,
      }
    }
    testID="checkbox"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "#43484d",
            "fontWeight": "normal",
            "marginLeft": 10,
            "marginRight": 10,
            "textTransform": "capitalize",
          }
        }
        testID="checkboxTitle"
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        OK
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Checkbox /> renders correctly when active 1`] = `
<View
  style={
    Object {
      "flexBasis": "50%",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#ededed",
        "borderRadius": 3,
        "borderWidth": 0,
        "margin": 5,
        "marginLeft": 10,
        "marginRight": 10,
        "opacity": 1,
        "padding": 0,
      }
    }
    testID="checkbox"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#2a9e91",
              "fontSize": 24,
            },
            Object {
              "minWidth": 24,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ÔÅÜ
      </Text>
      <Text
        style={
          Object {
            "color": "#43484d",
            "fontWeight": "normal",
            "marginLeft": 10,
            "marginRight": 10,
            "textTransform": "capitalize",
          }
        }
        testID="checkboxTitle"
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        OK
      </Text>
    </View>
  </View>
</View>
`;
